### ISSUE TITLE
Upgrade from 1.20.0-rc0 to 1.20.0 fails: Deprecated fallback present

### ISSUE THREAD

**Issue by hkdd** [go-gitea/gitea#25915]
### Description

I just did a docker pull of :latest and docker-compose down/up the container and I get this.  I was able to resolve by pinning it to 1.20.0-rc0 but 1.20.0-rc1 and rc2 failed as well as :latest which is currently 1.20.0.

Attaching to gitea_web                                                                                                         
gitea_web              | 2023/07/16 22:51:24 ...s/setting/setting.go:180:loadRunModeFrom() [F] Gitea is not supposed to be run a
s root. Sorry. If you need to use privileged TCP ports please instead use setcap and the `cap_net_bind_service` permission     
gitea_web              | Server listening on :: port 22.                                                                       
gitea_web              | Server listening on 0.0.0.0 port 22.                                                                  
gitea_web              | 2023/07/16 22:51:24 ...g/config_provider.go:327:deprecatedSettingFatal() [F] Deprecated fallback `[serv
er]` `LFS_CONTENT_PATH` present. Use `[lfs]` `PATH` instead. This fallback will be/has been removed in v1.19.0                 
gitea_web              | Received signal 15; terminating.                                                                      
gitea_web              | 2023/07/16 22:51:24 ...s/setting/setting.go:180:loadRunModeFrom() [F] Gitea is not supposed to be run a
s root. Sorry. If you need to use privileged TCP ports please instead use setcap and the `cap_net_bind_service` permission     
gitea_web              | Server listening on :: port 22.                                                                       
gitea_web              | Server listening on 0.0.0.0 port 22.                                                                  
gitea_web              | 2023/07/16 22:51:25 ...g/config_provider.go:327:deprecatedSettingFatal() [F] Deprecated fallback `[serv
er]` `LFS_CONTENT_PATH` present. Use `[lfs]` `PATH` instead. This fallback will be/has been removed in v1.19.0                 
gitea_web              | Received signal 15; terminating.                                                                      
gitea_web              | 2023/07/16 22:51:25 ...s/setting/setting.go:180:loadRunModeFrom() [F] Gitea is not supposed to be run a
s root. Sorry. If you need to use privileged TCP ports please instead use setcap and the `cap_net_bind_service` permission     
gitea_web              | Server listening on :: port 22.                                                                       
gitea_web              | Server listening on 0.0.0.0 port 22.                                                                  
gitea_web              | 2023/07/16 22:51:25 ...g/config_provider.go:327:deprecatedSettingFatal() [F] Deprecated fallback `[serv
er]` `LFS_CONTENT_PATH` present. Use `[lfs]` `PATH` instead. This fallback will be/has been removed in v1.19.0                 
gitea_web              | Received signal 15; terminating.                                                                      
gitea_web              | 2023/07/16 22:51:26 ...s/setting/setting.go:180:loadRunModeFrom() [F] Gitea is not supposed to be run a
s root. Sorry. If you need to use privileged TCP ports please instead use setcap and the `cap_net_bind_service` permission     
gitea_web              | Server listening on :: port 22.                                                                       
gitea_web              | Server listening on 0.0.0.0 port 22.                                                                  
gitea_web              | 2023/07/16 22:51:26 ...g/config_provider.go:327:deprecatedSettingFatal() [F] Deprecated fallback `[serv
er]` `LFS_CONTENT_PATH` present. Use `[lfs]` `PATH` instead. This fallback will be/has been removed in v1.19.0                 
gitea_web              | Received signal 15; terminating.                                     

### Gitea Version

1.20.0

### Can you reproduce the bug on the Gitea demo site?

Yes

### Git Version

1.20.0

### Operating System

EL8

### How are you running Gitea?

docker

### Database

None

**Comment by delvh** [comment]
The error message tells you explicitly what you need to do:
> Deprecated fallback `[server].LFS_CONTENT_PATH` present. Use `[lfs].PATH` instead. This fallback has been removed in v1.19.0

So, go to your `app.ini` and change that.

**Comment by eallion** [comment]
same

**Comment by stoofz** [comment]
Same issue as above in regards to updated Docker

**Comment by x86dev** [comment]
Just also reproduced this -- pulling the 1.20 container breaks, whereas pinning to 1.20.0-rc0 makes it work again. So something screwed up in the built container.

**Comment by realies** [comment]
what a weird change

**Comment by x86dev** [comment]
That fix does not work if you use the default container with no custom app.ini.

**Comment by GhaziTriki** [comment]
Working after deprecated configuration update.

**Comment by x86dev** [comment]
The cause and the solution is clear, but the container image just isn't up-to-date ;-)

**Comment by delvh** [comment]
Yes. This warning is explicitly meant to notify admins that such config options no longer have any effect.
This prevents the situation that settings are no longer in effect that admins thought they set. 
That's why this warning is crashing Gitea as intended when the setting is present.

After all, there are three ways to handle breaking config changes (that I see as viable):

1. Annoy admins slightly whenever they upgrade to fix their now outdated config to keep up to date
2. Carry old config options for quite a while and annoy admins **a lot** by removing all legacy options for several releases at once
3. Carry all old config options forever, and make the code effectively unmaintainable and unpredictable

Gitea currently chooses option `1.`.
I didn't go into detail about this error message previously in this issue as I thought my previous comment would suffice to let you know what happened, and how you can fix it.
Apparently I was wrong here. Sorry about that.

Regarding the Docker users:
If I've understood it correctly, there seems to be another issue in that the default Docker file comes with an app.ini that still sets a removed setting, right?
If that's the case, then that's a bug we should fix.

**Comment by TWDickson** [comment]
> Regarding the Docker users:
If I've understood it correctly, there seems to be another issue in that the default Docker file comes with an app.ini that still sets a removed setting, right?
If that's the case, then that's a bug we should fix

That’s my exact scenario, my docker-compose has no app.ini supplied and the /data directory is just set to a docker volume.

**Comment by delvh** [comment]
What config option is breaking the docker container exactly?
I can't find `[server].LFS_CONTENT_PATH` inside https://github.com/go-gitea/gitea/blob/ad4c09b59f1657a5400c92913972e4cca578a4fb/docker/root/etc/templates/app.ini, and the last commit on this file indicated that the setting was renamed, and that commit was 18 months ago.

**Comment by stoofz** [comment]
I guess the old app.ini in the docker container has to be manually edited. I rolled back the docker container to version 1.20.0-rc0 to be able to boot it up, then went into it via a console and manually added the [lfs] section and moved the PATH there while removing it from the [server] section. Now I am able to update and use the current docker image.

[lfs]
PATH=/data/git/lfs

[server]
APP_DATA_PATH=/data/gitea
...
LFS_START_SERVER=true
LFS_JWT_SECRET=xxx

**Comment by KN4CK3R** [comment]
You can edit the files in the volume even if the container is not running.

**Comment by chascode** [comment]
Does `LFS_START_SERVER` and `LFS_JWT_SECRET` still belong in the `[server]` section?

**Comment by timbze** [comment]
> What config option is breaking the docker container exactly? I can't find `[server].LFS_CONTENT_PATH` inside https://github.com/go-gitea/gitea/blob/ad4c09b59f1657a5400c92913972e4cca578a4fb/docker/root/etc/templates/app.ini, and the last commit on this file indicated that the setting was renamed, and that commit was 18 months ago.

The setting is there for containers that have been created before the setting was removed. I finally found `/var/lib/docker/volumes/gitea_gitea/_data/gitea/conf/app.ini`. I removed the setting and now I can start the container.

**Comment by Peter-Hageman** [comment]
changing app.ini didn't help. going back to 1.18.0 did :(

**Comment by delvh** [comment]
Why not?
Were there additional fatal warnings?
Or did you modify the wrong config file?

**Comment by lunny** [comment]
I think this could be closed since #26094 merged which will be in the following v1.20.2 after #26154 merge.
